docs/pages/guides/using-sentry.mdx

```diff
@@ -16,20 +18,199 @@ import { CODE } from '~/ui/components/Text';
+
+<ConfigReactNative>
+
+If you do not use [Continuous Native Generation (CNG)](https://docs.expo.dev/workflow/continuous-native-generation/), then you should use the [`@sentry/wizard`](https://docs.sentry.io/platforms/react-native/#install).
+
+</ConfigReactNative>
+
+</Step>
+
+<Step label="4">
+
+### Update Metro configuration
+
+Sentry hooks into Metro to inject a "debug ID" into your source maps. This debug ID is used to associate source maps with releases. To enable this, you need to add the following to your **metro.config.js** file (if you don't have the file yet, create it in the root of your project):
+
+```js
+// This replaces `const { getDefaultConfig } = require('expo/metro-config');`
+const { getSentryExpoConfig } = require('@sentry/react-native/metro');
+
+// This replaces `const config = getDefaultConfig(__dirname);`
+const config = getSentryExpoConfig(__dirname);
+
+module.exports = config;
+```
+
+</Step>
+
+<Step label="5">
+
+### Initialize Sentry
+
+Add the following to your app's main file such as **App.js**:
+
+```js
+import * as Sentry from '@sentry/react-native';
+
+Sentry.init({
+  dsn: 'YOUR DSN HERE',
+  debug: true, // If `true`, Sentry will try to print out useful debugging information if something goes wrong with sending the event. Set it to `false` in production
+});
+```
+
+Now wrap the root component of your app with Sentry. This may be **App.js** or **index.js** depending on your project, or in [**app/\_layout.js** with Expo Router](https://docs.expo.dev/router/layouts/#create-a-layout-route).
+
+```js
+import * as Sentry from '@sentry/react-native';
+
+// Your App component here
+
+export default Sentry.wrap(App);
+```
+
+</Step>
+
+<Step label="6">
+
+### Verify the configuration
+
+Create a new release build of your app and verify that it uploads source maps correctly. You may want to add a button in your app to test that it is working and sourcemaps are wired up as expected, for example:
+
+```jsx
+import { Button } from 'react-native';
+
+// Inside some component
+<Button title="Press me" onPress={() => throw new Error('Hello, Sentry!')} />;
+```
+
+</Step>
+
+## Usage with EAS Build
+
+Ensure that `SENTRY_AUTH_TOKEN` is set in your build environment, and Sentry will automatically upload source maps for you. If you use environment variables rather than properties in your app config, ensure that those are set as well.
+
+Using the above instructions, no additional work is needed to integrate Sentry into your project when using EAS Build.
+
+## Usage with EAS Update
+
+### Tag configuration
+
+Configuring Sentry to tag your scope with information about your update allows to see errors happening on certain updates in Sentry dashboard.
+
+Add the following snippet in the global scope as early as possible in your application's lifecycle.
+
+```js
+import * as Updates from 'expo-updates';
+
+const { metadata, extra } = Updates.manifest;
```

Comment:
cc @quinlanj

Reply:
How about this: https://github.com/expo/expo/pull/25712#discussion_r1449709860

Open to suggestions for this suggestion 😄

Reply:
Should be able to disambiguate the subtype by checking for the presence of `metadata`: https://github.com/expo/expo/blob/main/apps/expo-go/src/menu/DevMenuTaskInfo.tsx#L28

Something like:
```typescript
const { metadata, extra } =
    'metadata' in manifest ? manifest : { metadata: undefined, extra: undefined };
```
or this might be cleaner:
```typescript
const metadata = 'metadata' in manifest ? manifest.metadata : undefined;
const extra = 'metadata' in manifest ? manifest.extra : undefined;
```

`metadata` in the manifest has type `object` and `extra` has type `ManifestExtra`.

Reply:
based on will's feedback: https://github.com/expo/expo/pull/25712#discussion_r1450825925

In this snippet, `updateGroup` is typed as unknown, so we do a runtime check for 'string' where necessary

Reply:
would this also preserve types?

```
const { manifest } = Updates;
const metadata = manifest?.metadata;
const extra = metadata ? manifest.extra : undefined;
const updateGroup = metadata?.updateGroup;
```

Reply:
Can't we use, or cast to the exported `Manifest` type?
* https://github.com/expo/expo/blob/ab91257ace9c5a8196eacf6f7e2391f659538c15/packages/expo-updates/src/Updates.types.ts#L3

Or we can discern which exact type it is, by checking the presence of i.e. `metadata` key, via `in` operator before casting, if needed. So:

```tsx
const typedManifest = 'metadata' in manifest ? (manifest as ExpoUpdatesManifest) : (manifest as EmbeddedManifest); 
```

Reply:
this seems like a good suggestion @Simek. up to @quinlanj / @wschurman

Reply:
It's also possible that it is empty object, though it's somewhat hard to tell when that's the case. Looks like it's when the library is disabled, either explicitly or due to an initialization error. So the actual type of `Updates.manifest` is `ExpoUpdatesManifest | EmbeddedManifest | {}` (I'll file a task to change it to not use `Partial<Manifest>`).